### ISSUE TITLE
Add more tests for CCLabelTTF

### ISSUE THREAD

**Issue by slembcke** [cocos2d/cocos2d-objc#791]
It seems only unstyled labels work for sure. Need to ui-tests for:
- [x] Drop shadow
- [x] Outline
- [x] Font color
- [x] Unstyled labels

Since different OS versions and platforms have different code paths, must test all on Mac, iOS 5 and 7.

**Comment by slembcke** [comment]
I found that shadow offsets were unaffected by the drawing transform and fixed that. Have not yet been able to reproduce any other issues.

I tried using a custom TTF font in SpriteBuilder like @jtwigg had suggested, but I can't actually get it to show up. When I select it as the font for a label I just get Helvetica. Unsure if it's a SB or Cocos bug yet.

**Comment by slembcke** [comment]
Seems like it just doesn't like Bitstream Vera Mono. Haven't figured out why. Trying a different font with SB worked, but I'm still unable to reproduce any label issues (other than the flipped shadow offset).

**Comment by vlidholt** [comment]
The file of the TTF font needs to have the exact same name as the font itself, or it will not show up in Cocos. This is probably something we should try to fix at some point, but not super urgent.

**Comment by slembcke** [comment]
Hmm. Good to know, I would have never guessed that.

Are there any other label flipping bugs? I'm wondering if it was an issue that was already fixed in Cocos but not merged into SpriteBuilder?

**Comment by vlidholt** [comment]
Just pulled the latest Cocos from develop branch. Labels are not showing up in SpriteBuilder (effects or not).

**Comment by slembcke** [comment]
Tried a clean checkout of SpriteBuilder and still can't reproduce the issue. Both styled and unstyled text works including a random custom TTF font. I triple checked that it was actually the latest Cocos code.

![labels](http://files.slembcke.net/upshot/upshot_GpwGpu1W.png)

I've poked around at the font settings a lot and can't find any combination of settings that breaks anything. Yesterday I spent quite a while adding and checking breakpoints to the TTF font code to make sure I wasn't missing any branches. No clue what else could be wrong.
